redbot/cogs/mod/kickban.py

```diff
@@ -638,16 +640,21 @@ async def tempban(
         async with self.config.guild(guild).current_tempbans() as current_tempbans:
             current_tempbans.append(member.id)
 
-        with contextlib.suppress(discord.HTTPException):
-            # We don't want blocked DMs preventing us from banning
-            msg = _("You have been temporarily banned from {server_name} until {date}.").format(
-                server_name=guild.name, date=unban_time.strftime("%m-%d-%Y %H:%M:%S")
-            )
-            if invite:
-                msg += _(" Here is an invite for when your ban expires: {invite_link}").format(
-                    invite_link=invite
-                )
-            await member.send(msg)
+        if settings["dm_on_kickban"]:
```

Comment:
The message should be sent *always* but the reason should only be sent *if* dm_on_kickban is True